RecoLocalCalo/HcalRecAlgos/src/HFSimpleTimeCheck.cc

```diff
@@ -25,14 +34,35 @@ unsigned HFSimpleTimeCheck::determineAnodeStatus(
     const unsigned nRaw = anode.nRaw();
     if (nRaw)
     {
-        const QIE10DataFrame::Sample s(anode.getRaw(nRaw - 1));
-        if (!s.ok())
+        bool hardwareOK = true;
+        unsigned sampleToCheck = anode.soi();
+        if (sampleToCheck < nRaw)
+        {
+            const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck));
+            hardwareOK = s.ok();
+        }
+        else
+        {
+            // This should not normally happen, but we still
+            // need to do something reasonable here
+            for (sampleToCheck = 0; sampleToCheck < nRaw; ++sampleToCheck)
+            {
+                const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck));
+                if (!s.ok())
+                    hardwareOK = false;
+            }
+        }
+        if (!hardwareOK)
```

Comment:
Sorry, which code duplication? I prefer to write my code
so that the least possible amount of work is performed
in the branch that is encountered most often.

On 06/27/2016 07:17 AM, Carl Vuosalo wrote:
>
> In RecoLocalCalo/HcalRecAlgos/src/HFSimpleTimeCheck.cc <https://github.com/cms-sw/cmssw/pull/14920#discussion_r68566046>:
>
> > +        {
> > +            const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck));
> > +            hardwareOK = s.ok();
> > +        }
> > +        else
> > +        {
> > +            // This should not normally happen, but we still
> > +            // need to do something reasonable here
> > +            for (sampleToCheck = 0; sampleToCheck < nRaw; ++sampleToCheck)
> > +            {
> > +                const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck));
> > +                if (!s.ok())
> > +                    hardwareOK = false;
> > +            }
> > +        }
> > +        if (!hardwareOK)
>
> Code duplication can be avoided here, starting at line 39:
>
> |unsigned startSample = sampleToCheck, endSample = sampleToCheck + 1; if (sampleToCheck >= nRaw) { startSample = 0; endSample = nRaw; } for (sampleToCheck = startSample; sampleToCheck < endSample; 
> ++sampleToCheck) { const QIE10DataFrame::Sample s(anode.getRaw(sampleToCheck)); if (!s.ok()) hardwareOK = false; } |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/14920/files/d69aae5b5dd9e76e1ba10f46e8ed58dd93044926#r68566046>, or mute the thread 
> <https://github.com/notifications/unsubscribe/AFS4-S7Y0LuphBdB-mu-lOI5IQC3_rpDks5qP79IgaJpZM4I5Mg7>.
>